### PR TITLE
ARROW-106 Support PyArrow 8.0

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

**Review by ShaneHarvey — APPROVED** [review]

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]
Wheels are failing with:
```
ValueError: Cannot repair wheel, because required library "libparquet.so.800" could not be located
```

I suppose parquet is a new dependency  added in Arrow 8 that we need to handle.

**Comment by blink1073** [comment]
> I suppose parquet is a new dependency added in Arrow 8 that we need to handle.

There is a parquet binary in the folder, it just isn't in `pa.get_libraries()`.  I added some special handling.

**Comment by ShaneHarvey** [comment]
> There is a parquet binary in the folder, it just isn't in pa.get_libraries()

Should we report that as a bug?

File: bindings/python/setup.py [hunk 1]
```diff
@@ -158,11 +158,11 @@ def append_arrow_flags(module):
         pass
 
     # Find the appropriate library file and optionally copy it locally.
-    for name in pa.get_libraries():
+    for name in pa.get_libraries() + ["parquet"]:
         if IS_WIN:
             if COPY_LIBARROW:
                 lib_file = os.path.join(arrow_lib, f"{name}.dll")
-                if not os.path.exists(lib_file):
+                if not os.path.exists(lib_file) and name != "parquet":
```
Comment (ShaneHarvey) [inline bindings/python/setup.py:165]: Won't shutil.copy fail when the parquet lib_file path doesn't exist?

Comment (blink1073) [reply bindings/python/setup.py:165]: Fixed

**Comment by blink1073** [comment]
> Should we report that as a bug?

https://issues.apache.org/jira/browse/ARROW-17327

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**